AssetRipper.Library/Exporters/Scripts/Transforms/EnsureValidBaseConstructorTransform.cs

```diff
@@ -150,6 +150,12 @@ public override void VisitTypeDeclaration(TypeDeclaration typeDeclaration)
 
 				if (TryGetBestConstructor(type, constructorDeclaration, out bestConstructor, out bool isBaseConstructor))
 				{
+					// fix for error CS0768: Constructor '<name>' cannot call itself through another constructor
+					if (!isBaseConstructor)
+					{
+						continue;
```

Comment:
I originally added support for this constructor calls because the original goal was to try to guess constructors that called themselves and had an extra parameter. Example:

```cs
public class MyAttribute : PropertyAttribute {
    public MyAttribute(string title) : base()
    { }
    public MyAttribute(string title, string label) : this(title)
    { }
}
```

IMO `TryGetBestConstructor` should be reworked to not attempt to find constructors on the original type at all.

Also you said you are dealing with a Mono game? In which case, this transform shouldn't run at all unless I missed a PR.